new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java

```diff
@@ -1406,8 +1446,87 @@ public static V1ServiceList listServices(String namespace) {
     }
   }
 
-  // --------------------------- jobs ---------------------------
+  /**
+   * Get V1Service object for the given service name, label and namespace.
+   *
+   * @param serviceName name of the service to look for
+   * @param label  key value pair with which the service is decorated with
+   * @param namespace namespace in which to check for the service
+   * @return V1Service object if found otherwise null
+   * @throws ApiException when there is error in querying the cluster
+   */
+  public static V1Service getService(
+      String serviceName, Map<String, String> label, String namespace)
+      throws ApiException {
+    String labelSelector = null;
+    if (label != null) {
+      String key = label.keySet().iterator().next().toString();
+      String value = label.get(key).toString();
+      labelSelector = String.format("%s in (%s)", key, value);
+      logger.info(labelSelector);
+    }
+    V1ServiceList v1ServiceList
```

Comment:
there is a name spaced list method for service. So you don't need to iterate through the service object to match the  namespace.
coreV1Api.listNamespacedService(......)

Reply:
Anyhow I have to loop thru all the service(s)  in a  namesapce even if i use coreV1Api.listNamespacedService(...)

Reply:
Yes , but one less check for namespace